python/benchmark/test_gen_data.py

```diff
@@ -159,19 +165,27 @@ def test_make_regression(dtype: str, low_rank: bool, use_gpu: str) -> None:
         assert c.shape == (10,), "coef shape mismatch"
         assert sum(c != 0.0) == 3, "Unexpected number of informative features"
 
-        # Test that y ~= np.dot(X, c) + bias + N(0, 1.0).
-        assert_almost_equal(np.std(y - np.dot(X, c)), 1.0, decimal=1)
+        if logistic_regression == "True":
+            # Test that X is consist of only 0 or 1
+            for n in y:
+                assert n == 0 or n == 1
+        else:
+            # Test that y ~= np.dot(X, c) + bias + N(0, 1.0).
+            assert_almost_equal(np.std(y - np.dot(X, c)), 1.0, decimal=1)
 
 
-@pytest.mark.parametrize("dtype", ["float32", "float64"])
-@pytest.mark.parametrize("low_rank", [True, False])
+@pytest.mark.parametrize("dtype", ["float64"])
 @pytest.mark.parametrize("use_gpu", ["True", "False"])
-def test_make_logistic_regression(dtype: str, low_rank: bool, use_gpu: str) -> None:
+@pytest.mark.parametrize("redundant_cols", ["0", "2"])
+@pytest.mark.parametrize("logistic_regression", ["True", "False"])
+def test_make_sparse_regression(
+    dtype: str, use_gpu: str, redundant_cols: str, logistic_regression: str
+) -> None:
     input_args = [
         "--num_rows",
         "100",
         "--num_cols",
-        "10",
+        "20",
```

Comment:
Please test at least two combinations of (num_rows, num_cols). For example (100, 20) and (10000, 55).